src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java

```diff
@@ -372,4 +386,55 @@ public void deleteInventory(MedicalInventory medicalInventory) throws OHServiceE
 		}
 		ioOperations.deleteInventory(medicalInventory);
 	}
+
+	/**
+	 * Confirm the Inventory rows of inventory.
+	 *
+	 * @param inventory - The {@link MedicalInventory}
+	 * @param inventoryRowSearchList- The list of {@link MedicalInventory}
+	 * @throws OHDataValidationException
+	 */
+	@Transactional(rollbackFor = OHServiceException.class)
+	public void confirmMedicalInventoryRow(MedicalInventory inventory, List<MedicalInventoryRow> inventoryRowSearchList) throws OHServiceException {
+		// validate the inventory
+		this.validateMedicalInventoryRow(inventory, inventoryRowSearchList);
+		// create Movements
+		String dischargeCode = inventory.getDischargeType();
+		String chargeCode = inventory.getChargeType();
+		Integer supplierId = inventory.getSupplier();
+		String wardCode = inventory.getDestination();
+		MovementType chargeType = medicalDsrStockMovementTypeBrowserManager.getMovementType(chargeCode);
+		MovementType dischargeType = medicalDsrStockMovementTypeBrowserManager.getMovementType(dischargeCode);
+		Supplier supplier = supplierManager.getByID(supplierId);
+		Ward ward = wardManager.findWard(wardCode);
+		LocalDateTime now = TimeTools.getNow();
+		for (Iterator<MedicalInventoryRow> iterator = inventoryRowSearchList.iterator(); iterator.hasNext();) {
+			MedicalInventoryRow medicalInventoryRow = (MedicalInventoryRow) iterator.next();
+			StringBuilder referenceBuilder = new StringBuilder();
+			referenceBuilder.append(inventory.getInventoryReference())
+			                .append("-")
+			                .append(now);
+			double theoQty = medicalInventoryRow.getTheoreticQty();
+			double realQty = medicalInventoryRow.getRealQty();
+			Double ajustQty = theoQty - realQty;
+			Medical medical = medicalInventoryRow.getMedical();
+			String lotCode = medicalInventoryRow.getLot().getCode();
+			Lot currentLot = movStockInsertingManager.getLot(lotCode);
+			if (realQty > theoQty) { // charge movement when realQty > theoQty
+				referenceBuilder.append("-charge");
```

Comment:
this would append a "-charge" at each cycle.... I think we should just define a `chargeReference` and a `dischargeReference` outside the iteration cycle

Reply:
Moreover, since we are adding two suffix, we have to check that the reference from the user does not exceed 50 chars, as MMV_REFNO is defined as VARCHAR(50)